src/views/Drive/components/VersionHistory/services/file-version.service.ts

```diff
@@ -0,0 +1,61 @@
+import { SdkFactory } from 'app/core/factory/sdk';
+import localStorageService from 'services/local-storage.service';
+import { downloadFile } from 'app/network/download';
+import { binaryStreamToBlob } from 'services/stream.service';
+import { saveAs } from 'file-saver';
+import { FileVersion } from '../types';
+
+const getStorageClient = () => SdkFactory.getNewApiInstance().createNewStorageClient();
+
+export async function getFileVersions(fileUuid: string): Promise<FileVersion[]> {
+  return getStorageClient().getFileVersions(fileUuid);
+}
+
+export async function deleteVersion(fileUuid: string, versionId: string): Promise<void> {
+  await getStorageClient().deleteFileVersion(fileUuid, versionId);
+}
+
+export async function restoreVersion(fileUuid: string, versionId: string): Promise<FileVersion> {
+  return getStorageClient().restoreFileVersion(fileUuid, versionId);
+}
+
+export async function downloadVersion(
+  version: FileVersion,
+  fileName: string,
+  bucketId: string,
+  updateProgressCallback?: (progress: number) => void,
+  abortController?: AbortController,
+): Promise<void> {
+  const user = localStorageService.getUser();
+  if (!user) throw new Error('User not found');
+
+  const fileStream = await downloadFile({
```

Comment:
Can we use the multipart download for this? Imagine the file has 5GB or more. It would be better to download it in parallel-small chunks instead of the full file at once. You can do it by using the `DownloadManager.downloadItem` if I'm not mistaken.